### ISSUE TITLE
[NPC][WoTLK] Goretalon Roc and Shattertusk Bull, missing paths, wrong state

### ISSUE THREAD

**Issue by Teridan-dal-wow** [dalaranwow/dalaran-wow#4492]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Current behaviour**: Npc http://wotlk.openwow.com/npc=28004 is missing all its paths and npc http://wotlk.openwow.com/npc=28380 has the wrong state and be part of a mini event.

**Expected behaviour**: Npc http://wotlk.openwow.com/npc=28004 should have paths and fly around, while http://wotlk.openwow.com/npc=28380 should be dead all the time and both of them are part of mini event.
Video of the event from retail legion from 27.02.2017:
https://www.youtube.com/watch?v=pIyrD8eCsfg

What is done:
- added the missing paths
- fixed the state of the bulls
- removed random movement from all of these creatures
- added fly state for all rocs

To do:
- add the mini event (roc goes near dead bull and casts 3 times (with small delay between each cast) spell http://wotlk.openwow.com/spell=52930 then returns to it's path

How it looks now:
![image](https://cloud.githubusercontent.com/assets/6797611/23368740/80d7005a-fd17-11e6-8db9-084f8ecf22dd.png)
![image](https://cloud.githubusercontent.com/assets/6797611/23368756/9075ad9a-fd17-11e6-97b5-588e68dcf8de.png)

``` sql
UPDATE `creature` SET `spawndist` = 0 , `MovementType` = 0 WHERE `guid` IN (106809, 104810, 104816, 104818, 104822, 104823, 104825, 104826, 104833, 104834, 104835, 104862); 
UPDATE `creature` SET `spawndist` = 0 , `MovementType` = 0 WHERE `guid` BETWEEN 104838 AND 104853;

UPDATE `creature` SET `spawndist` = 0 , `MovementType` = 0, `unit_flags` = 33536 WHERE `guid` IN (116995, 117008, 116984, 116997, 116969, 116994, 116983);
UPDATE `creature_template` SET `InhabitType` = 4 WHERE `entry` IN (28004, 29044); 
UPDATE `creature_addon` SET `bytes1` = 1 WHERE `guid` = 104818; 

DELETE FROM `creature_addon` WHERE `guid` IN (116995, 117008, 116984, 116997, 116969, 116994, 116983);
INSERT INTO `creature_addon` (`guid`, `path_id`, `mount`, `bytes1`, `bytes2`, `emote`, `auras`) VALUES
(116995, 0, 0, 0, 1, 0, '29266'), -- 28380 - 29266 - 29266
(117008, 0, 0, 0, 1, 0, '29266'), -- 28380 - 29266 - 29266
(116984, 0, 0, 0, 1, 0, '29266'), -- 28380 - 29266 - 29266
(116997, 0, 0, 0, 1, 0, '29266'), -- 28380 - 29266 - 29266
(116969, 0, 0, 0, 1, 0, '29266'), -- 28380 - 29266 - 29266
(116994, 0, 0, 0, 1, 0, '29266'), -- 28380 - 29266 - 29266
(116983, 0, 0, 0, 1, 0, '29266'); -- 28380 - 29266 - 29266


SET @NPC := 104849;
SET @PATH := @NPC * 10;
UPDATE `creature` SET `spawndist`=0,`MovementType`=2,`position_x`=6502.640,`position_y`=4529.336,`position_z`=-39.647 WHERE `guid`=@NPC;
DELETE FROM `creature_addon` WHERE `guid`=@NPC;
INSERT INTO `creature_addon` (`guid`,`path_id`,`mount`,`bytes1`,`bytes2`,`emote`,`auras`) VALUES (@NPC,@PATH,0,0,1,0, '');
DELETE FROM `waypoint_data` WHERE `id`=@PATH;
INSERT INTO `waypoint_data` (`id`,`point`,`position_x`,`position_y`,`position_z`,`orientation`,`delay`,`move_type`,`action`,`action_chance`,`wpguid`) VALUES
(@PATH,1,6502.640,4529.336,-39.647,0,0,1,0,100,0), -- 17:00:29
(@PATH,2,6519.066,4473.544,-39.647,0,0,1,0,100,0), -- 17:00:42
(@PATH,3,6574.749,4492.481,-39.647,0,0,1,0,100,0), -- 17:00:40
(@PATH,4,6559.046,4563.689,-39.647,0,0,1,0,100,0), -- 17:00:00
(@PATH,5,6520.473,4557.068,-39.647,0,0,1,0,100,0); -- 16:59:59

SET @NPC := 104835;
SET @PATH := @NPC * 10;
UPDATE `creature` SET `spawndist`=0,`MovementType`=2 WHERE `guid`=@NPC;
DELETE FROM `creature_addon` WHERE `guid`=@NPC;
INSERT INTO `creature_addon` (`guid`,`path_id`,`mount`,`bytes1`,`bytes2`,`emote`,`auras`) VALUES (@NPC,@PATH,0,0,1,0, '');
DELETE FROM `waypoint_data` WHERE `id`=@PATH;
INSERT INTO `waypoint_data` (`id`,`point`,`position_x`,`position_y`,`position_z`,`orientation`,`delay`,`move_type`,`action`,`action_chance`,`wpguid`) VALUES
(@PATH,1,6407.776,4398.972,-28.775,0,0,1,0,100,0), -- 17:00:29
(@PATH,2,6457.820,4375.584,-28.775,0,0,1,0,100,0), -- 17:00:42
(@PATH,3,6489.897,4417.942,-28.775,0,0,1,0,100,0), -- 17:00:40
(@PATH,4,6430.807,4446.0317,-28.775,0,0,1,0,100,0);

SET @NPC := 104816;
SET @PATH := @NPC * 10;
UPDATE `creature` SET `spawndist`=0,`MovementType`=2 WHERE `guid`=@NPC;
DELETE FROM `creature_addon` WHERE `guid`=@NPC;
INSERT INTO `creature_addon` (`guid`,`path_id`,`mount`,`bytes1`,`bytes2`,`emote`,`auras`) VALUES (@NPC,@PATH,0,0,1,0, '');
DELETE FROM `waypoint_data` WHERE `id`=@PATH;
INSERT INTO `waypoint_data` (`id`,`point`,`position_x`,`position_y`,`position_z`,`orientation`,`delay`,`move_type`,`action`,`action_chance`,`wpguid`) VALUES
(@PATH,1,6436.410,4575.279,-47.262,0,0,1,0,100,0), -- 17:00:29
(@PATH,2,6434.633,4534.298,-47.262,0,0,1,0,100,0), -- 17:00:42
(@PATH,3,6501.483,4520.978,-47.262,0,0,1,0,100,0), -- 17:00:40
(@PATH,4,6490.995,4589.329,-47.262,0,0,1,0,100,0),
(@PATH,5,6454.477,4596.040,-47.262,0,0,1,0,100,0);

SET @NPC := 104852;
SET @PATH := @NPC * 10;
UPDATE `creature` SET `spawndist`=0,`MovementType`=2,`position_x`=6596.614,`position_y`=4373.054,`position_z`=15.631 WHERE `guid`=@NPC;
DELETE FROM `creature_addon` WHERE `guid`=@NPC;
INSERT INTO `creature_addon` (`guid`,`path_id`,`mount`,`bytes1`,`bytes2`,`emote`,`auras`) VALUES (@NPC,@PATH,0,0,1,0, '');
DELETE FROM `waypoint_data` WHERE `id`=@PATH;
INSERT INTO `waypoint_data` (`id`,`point`,`position_x`,`position_y`,`position_z`,`orientation`,`delay`,`move_type`,`action`,`action_chance`,`wpguid`) VALUES
(@PATH,1,6596.614,4373.054,15.631,0,0,1,0,100,0), -- 17:00:29
(@PATH,2,6545.000,4440.160,-23.21,0,0,1,0,100,0), -- 17:00:42
(@PATH,3,6507.370,4462.658,-23.21,0,0,1,0,100,0), -- 17:00:40
(@PATH,4,6492.722,4417.610,-23.21,0,0,1,0,100,0);

SET @NPC := 104822;
SET @PATH := @NPC * 10;
UPDATE `creature` SET `spawndist`=0,`MovementType`=2,`position_x`=6484.867,`position_y`=4423.562,`position_z`=-24.175 WHERE `guid`=@NPC;
DELETE FROM `creature_addon` WHERE `guid`=@NPC;
INSERT INTO `creature_addon` (`guid`,`path_id`,`mount`,`bytes1`,`bytes2`,`emote`,`auras`) VALUES (@NPC,@PATH,0,0,1,0, '');
DELETE FROM `waypoint_data` WHERE `id`=@PATH;
INSERT INTO `waypoint_data` (`id`,`point`,`position_x`,`position_y`,`position_z`,`orientation`,`delay`,`move_type`,`action`,`action_chance`,`wpguid`) VALUES
(@PATH,1,6484.867,4423.562,-24.175,0,0,1,0,100,0), -- 17:00:29
(@PATH,2,6426.345,4441.955,-24.175,0,0,1,0,100,0), -- 17:00:42
(@PATH,3,6380.181,4348.904,-24.175,0,0,1,0,100,0), -- 17:00:40
(@PATH,4,6521.990,4438.359,-24.175,0,0,1,0,100,0),
(@PATH,5,6526.768,4361.264,-24.175,0,0,1,0,100,0);

SET @NPC := 104839;
SET @PATH := @NPC * 10;
UPDATE `creature` SET `spawndist`=0,`MovementType`=2 WHERE `guid`=@NPC;
DELETE FROM `creature_addon` WHERE `guid`=@NPC;
INSERT INTO `creature_addon` (`guid`,`path_id`,`mount`,`bytes1`,`bytes2`,`emote`,`auras`) VALUES (@NPC,@PATH,0,0,1,0, '');
DELETE FROM `waypoint_data` WHERE `id`=@PATH;
INSERT INTO `waypoint_data` (`id`,`point`,`position_x`,`position_y`,`position_z`,`orientation`,`delay`,`move_type`,`action`,`action_chance`,`wpguid`) VALUES
(@PATH,1,6371.549,4249.790,-27.263,0,0,1,0,100,0), -- 17:00:29
(@PATH,2,6368.297,4311.977,-27.263,0,0,1,0,100,0), -- 17:00:42
(@PATH,3,6431.363,4323.808,-27.263,0,0,1,0,100,0), -- 17:00:40
(@PATH,4,6413.182,4254.311,-27.263,0,0,1,0,100,0);

SET @NPC := 104847;
SET @PATH := @NPC * 10;
UPDATE `creature` SET `spawndist`=0,`MovementType`=2,`position_x`=6511.695,`position_y`=4477.459,`position_z`=-31.453 WHERE `guid`=@NPC;
DELETE FROM `creature_addon` WHERE `guid`=@NPC;
INSERT INTO `creature_addon` (`guid`,`path_id`,`mount`,`bytes1`,`bytes2`,`emote`,`auras`) VALUES (@NPC,@PATH,0,0,1,0, '');
DELETE FROM `waypoint_data` WHERE `id`=@PATH;
INSERT INTO `waypoint_data` (`id`,`point`,`position_x`,`position_y`,`position_z`,`orientation`,`delay`,`move_type`,`action`,`action_chance`,`wpguid`) VALUES
(@PATH,1,6511.695,4477.459,-31.453,0,0,1,0,100,0), -- 17:00:29
(@PATH,2,6488.574,4531.824,-31.453,0,0,1,0,100,0), -- 17:00:42
(@PATH,3,6441.916,4525.009,-31.453,0,0,1,0,100,0), -- 17:00:40
(@PATH,4,6434.817,4457.507,-31.453,0,0,1,0,100,0),
(@PATH,5,6478.626,4454.020,-31.453,0,0,1,0,100,0);


```

**Comment by Teridan-dal-wow** [comment]
Maybe the SAI of the rocs should look something like this? (timing is not done and specific sai for guids)
``` sql
-- Goretalon Roc SAI
SET @ENTRY := 28004;
UPDATE `creature_template` SET `AIName`="SmartAI" WHERE `entry`=@ENTRY;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=0;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(@ENTRY,0,0,0,0,0,100,0,4000,7000,7000,10000,11,55079,0,0,0,0,0,2,0,0,0,0,0,0,0,"Goretalon Roc - In Combat - Cast 'Swoop' (No Repeat) (Dungeon)"),
(@ENTRY,0,1,0,1,0,100,0,1000,1000,20000,20000,80,@ENTRY*100+00,0,0,0,0,0,1,0,0,0,0,0,0,0,"Goretalon Roc - Out of Combat - Run Script");

-- Actionlist SAI
SET @ENTRY := 2800400;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=9;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(@ENTRY,9,0,5,0,0,100,0,0,0,0,0,48,1,0,0,0,0,0,1,0,0,0,0,0,0,0,"Goretalon Roc - On Script - Set Active On"),
(@ENTRY,9,5,6,0,0,100,0,0,0,0,0,54,15000,0,0,0,0,0,1,0,0,0,0,0,0,0,"Goretalon Roc - On Script - Pause Waypoint"),
(@ENTRY,9,6,7,0,0,100,0,0,0,0,0,69,0,0,0,0,0,0,19,28380,30,0,0,0,0,0,"Goretalon Roc - On Script - Move To Closest Creature 'Shattertusk Bull'"),
(@ENTRY,9,7,8,0,0,100,0,3500,3500,0,0,11,52930,0,0,0,0,0,1,0,0,0,0,0,0,0,"Goretalon Roc - On Script - Cast 'Feed'"),
(@ENTRY,9,8,9,0,0,100,0,2000,2000,0,0,11,52930,0,0,0,0,0,1,0,0,0,0,0,0,0,"Goretalon Roc - On Script - Cast 'Feed'"),
(@ENTRY,9,9,10,0,0,100,0,2000,2000,0,0,11,52930,0,0,0,0,0,1,0,0,0,0,0,0,0,"Goretalon Roc - On Script - Cast 'Feed'"),
(@ENTRY,9,10,0,0,0,100,0,0,0,0,0,48,0,0,0,0,0,0,1,0,0,0,0,0,0,0,"Goretalon Roc - On Script - Set Active Off");
```

**Comment by Rushor** [comment]
no we cannot use Move To Closest Creature together with waypoint_data currently. only pausing the waypoints works, anyways i will push the existing fixes

**Comment by Rushor** [comment]
DB/Creature: Goretalon Roc and Shattertusk Bull - Add missing paths and update state

* closes: https://github.com/dalaranwow/dalaran-wow/issues/4492
* commit: e420f31272afad1728c51819a98ad91023af5675

i will push the fix for the event as soon as i implemented a way to handle it. i thought about an option to store the current wp-id before the events starts and let the creature resume that point after the event is done

**Comment by Teridan-dal-wow** [comment]
> no we cannot use Move To Closest Creature together with waypoint_data currently. only pausing the waypoints works, anyways i will push the existing fixes

What do you mean? The roc does go down and lands at the bull.

**Comment by Rushor** [comment]
```sql
(@ENTRY,9,6,7,0,0,100,0,0,0,0,0,69,0,0,0,0,0,0,19,28380,30,0,0,0,0,0,"Goretalon Roc - On Script - Move To Closest Creature 'Shattertusk Bull'"),
```

This action will not work even if we pause the waypoint movement. MovePoint() function will directly init the waypoint_movement again even if it was stopped

**Comment by Teridan-dal-wow** [comment]
Hm, if that is a dead end, a possible workaround is to integrate the mini event in each roc's path as action on reach point X. The roc can make 2 circles around its path and go down for a "feed".

**Comment by Rushor** [comment]
yes yes, but we need to add that work-around for all events in the future. i'm tired of using hacks. i will fix it right :D i'm currently compiling an alpha version of storing and updating the currentwaypoint. let's see if it works fine :D

**Comment by Rushor** [comment]
alright this works so far:

```sql
-- Goretalon Roc SAI
SET @ENTRY := 28004;
UPDATE `creature_template` SET `AIName`="SmartAI" WHERE `entry`=@ENTRY;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=0;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(@ENTRY,0,0,0,0,0,100,0,4000,7000,7000,10000,11,55079,0,0,0,0,0,2,0,0,0,0,0,0,0,"Goretalon Roc - In Combat - Cast 'Swoop'"),
(@ENTRY,0,1,0,1,0,100,0,6000,6000,60000,60000,80,@ENTRY*100+00,0,0,0,0,0,1,0,0,0,0,0,0,0,"Goretalon Roc - Out of Combat - Run Script"),
(@ENTRY,0,2,0,34,0,100,0,8,5,0,0,80,@ENTRY*100+01,0,0,0,0,0,1,0,0,0,0,0,0,0,"Goretalon Roc - On Reached Point 5 - Run Script");

-- Only run Goretalon Roc OOC Event if creature has Movementtype Waypoint
DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=22 AND `SourceGroup`=2 AND `SourceEntry`=28004;
INSERT INTO `conditions` (`SourceTypeOrReferenceId`,`SourceGroup`,`SourceEntry`,`SourceId`,`ElseGroup`,`ConditionTypeOrReference`,`ConditionTarget`,`ConditionValue1`,`ConditionValue2`,`ConditionValue3`,`NegativeCondition`,`ErrorType`,`ErrorTextId`,`ScriptName`,`Comment`) VALUES
(22,2,28004,0,0,41,1,2,0,0,0,0,0,"","Only run Goretalon Roc OOC Event if creature has Movementtype Waypoint"),
(22,2,28004,0,0,29,1,28380,60,0,0,0,0,"","Only run Goretalon Roc OOC Event if creature is near Creature 'Shattertusk Bull'");

-- Actionlist SAI
SET @ENTRY := 2800400;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=9;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(@ENTRY,9,0,0,0,0,100,0,0,0,0,0,48,1,0,0,0,0,0,1,0,0,0,0,0,0,0,"Goretalon Roc - On Script - Set Active On"),
(@ENTRY,9,1,0,0,0,100,0,0,0,0,0,210,0,0,0,0,0,0,1,0,0,0,0,0,0,0,"Goretalon Roc - On Script - Store Current Waypoint ID"),
(@ENTRY,9,2,0,0,0,100,0,0,0,0,0,147,0,1,0,0,0,0,1,0,0,0,0,0,0,0,"Goretalon Roc - On Script - Movement Expired"),
(@ENTRY,9,3,0,0,0,100,0,0,0,0,0,59,1,0,0,0,0,0,1,0,0,0,0,0,0,0,"Goretalon Roc - On Script - Set Run On"),
(@ENTRY,9,4,0,0,0,100,0,0,0,0,0,69,5,0,0,0,0,0,19,28380,60,0,0,0,0,0,"Goretalon Roc - On Script - Move To Closest Creature 'Shattertusk Bull'");

-- Actionlist SAI
SET @ENTRY := 2800401;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=9;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(@ENTRY,9,0,0,0,0,100,0,2500,2500,0,0,11,52930,0,0,0,0,0,1,0,0,0,0,0,0,0,"Goretalon Roc - On Script - Cast 'Feed'"),
(@ENTRY,9,1,0,0,0,100,0,2000,2000,0,0,11,52930,0,0,0,0,0,1,0,0,0,0,0,0,0,"Goretalon Roc - On Script - Cast 'Feed'"),
(@ENTRY,9,2,0,0,0,100,0,2000,2000,0,0,11,52930,0,0,0,0,0,1,0,0,0,0,0,0,0,"Goretalon Roc - On Script - Cast 'Feed'"),
(@ENTRY,9,3,0,0,0,100,0,0,0,0,0,209,0,1,0,0,0,0,1,0,0,0,0,0,0,0,"Goretalon Roc - On Script - Load WP Path"),
(@ENTRY,9,4,0,0,0,100,0,0,0,0,0,211,0,0,0,0,0,0,1,0,0,0,0,0,0,0,"Goretalon Roc - On Script - Update Current Waypoint ID"),
(@ENTRY,9,5,0,0,0,100,0,0,0,0,0,48,0,0,0,0,0,0,1,0,0,0,0,0,0,0,"Goretalon Roc - On Script - Set Active Off");

DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=13 AND  `SourceGroup`=1 AND `SourceEntry`=52930;
INSERT INTO `conditions` (`SourceTypeOrReferenceId`, `SourceGroup`, `SourceEntry`, `SourceId`, `ElseGroup`, `ConditionTypeOrReference`, `ConditionTarget`, `ConditionValue1`, `ConditionValue2`, `ConditionValue3`, `NegativeCondition`, `ErrorType`, `ErrorTextId`, `ScriptName`, `Comment`) VALUES 
(13, 1, 52930, 0, 0, 31, 0, 3, 28380, 0, 0, 0, 0, '', 'Feed only target Shattertusk Bull');
```

there is one small issue left with soring the current waypoint id - i hope @robinsch can investigate it (i will open up a new issue for it)

**Comment by Rushor** [comment]
DB/Creature: Add Path and emoteevent for Goretalon Roc in Sholazar Basin

* closes: https://github.com/dalaranwow/dalaran-wow/issues/4492
* commit: 973cc3c20db5ce39e21f00fae2962f61301d4483

The issue was fixed by a developer and will now be tested on the Public Test Realm (PTR). The changes/fixes will be applied with the NEXT changelog on the liverealm.
This means that the issue exists in game at the moment i wrote this message.
Only once the issue was marked with "Fixed - On Live Server", you can find the fix on you server.
Please watch http://www.dalaran-wow.com/news/ to see when the changelog and the fix was applied.

**Comment by Rushor** [comment]
Okay the creatures now restart their path from point 1 again after the event is done. 
But they should continue their path from the last point instead. Fine Tuning for this will be make in https://github.com/dalaranwow/dalaran-wow/issues/4494

**Comment by Rushor** [comment]
Game/SmartScripts: Implement NextWaypointid in SMART_ACTION_SET_NEXT_WP_ID which uses the StoredWaypointID() + the value from NextWaypointid

* The resumed path will look more smooth now if we use StoredWaypointID()  + 1
* updates: https://github.com/dalaranwow/dalaran-wow/issues/4492
* commit: 5e6690a14c12720eb0542819ba2897a86648f017
